### PR TITLE
Design notification interface

### PR DESCRIPTION
**Pull request by yongsokheng**

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -16,4 +16,5 @@
 //= require jquery_nested_form
 //= require bootstrap-datepicker
 //= require bootstrap-sprockets
+//= require jquery.slimscroll
```
Comment (NgengChhengkim) [inline app/assets/javascripts/application.js]: require tree hoy co can require o day nua ko?

File: app/models/notification.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Notification < ApplicationRecord
+  belongs_to :user
+  belongs_to :receiver, polymorphic: true
```
Comment (NgengChhengkim) [inline app/models/notification.rb]: user lai ai? reciever la ai?

File: app/models/user_notification.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class UserNotification < ApplicationRecord
+  belongs_to :notification
+  belongs_to :user
+
+  scope :unseen, ->{where seen: false}
+  scope :load_notifications, ->{includes notification: :user}
```
Comment (NgengChhengkim) [inline app/models/user_notification.rb]: preload?

File: app/models/user_notification.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class UserNotification < ApplicationRecord
+  belongs_to :notification
+  belongs_to :user
+
+  scope :unseen, ->{where seen: false}
+  scope :load_notifications, ->{preload notification: :user}
```
Comment (NgengChhengkim) [inline app/models/user_notification.rb]: preload kroy limit

File: app/models/notification.rb (hunk 2 above)
Comment (yongsokheng) [reply app/models/notification.rb]: user la nguoi tao notification
receiver la cac doi tuong khac nhu post, like, comment, .... tuc la notification cua user tren receiver ay

File: app/models/user_notification.rb (hunk 4 above)
Comment (yongsokheng) [reply app/models/user_notification.rb]: ber cheng viet scope mech ket?

File: app/models/notification.rb (hunk 2 above)
Comment (NgengChhengkim) [reply app/models/notification.rb]: co ten khac ngoai reciever khong? doc xong user va reciever khong biet la gi

Comment (yongsokheng) [reply app/models/notification.rb]: dat ten la gi? receiver cung dung nghia roi. :smile:

File: app/models/user_notification.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+class UserNotification < ApplicationRecord
+  belongs_to :notification
+  belongs_to :user
+
+  scope :unseen, ->{where seen: false}
```
Comment (NgengChhengkim) [inline app/models/user_notification.rb]: seen co the doi dung integer duoc khong? dung enum. neu sau nay muon them status read nua se de

Comment (yongsokheng) [reply app/models/user_notification.rb]: seen? => lay tat ca seen, nhung cua minh thi lay tat ca chua seen ma. bay gio minh co ca 2 seen va read

File: app/models/notification.rb (hunk 2 above)
Comment (NgengChhengkim) [reply app/models/notification.rb]: nghia la notification cua Like , post, .... la notification cua reciever?

File: app/models/user_notification.rb (hunk 5 above)
Comment (NgengChhengkim) [reply app/models/user_notification.rb]: sao ko dung mot cai status?

File: app/models/user_notification.rb (hunk 4 above)
Comment (NgengChhengkim) [reply app/models/user_notification.rb]: preloadd khong can viet trong scope

File: app/models/notification.rb (hunk 2 above)
Comment (yongsokheng) [reply app/models/notification.rb]: the thi dat la trackable thi sao?

File: app/models/user_notification.rb (hunk 4 above)
Comment (NgengChhengkim) [reply app/models/user_notification.rb]: preload mun cheng load mun mok tang ors. te yerng load mok tam so luong limit ta

File: app/models/user_notification.rb (hunk 5 above)
Comment (yongsokheng) [reply app/models/user_notification.rb]: y la dung enum status: [:unseen, :seen, :read]?

File: app/models/user_notification.rb (hunk 4 above)
Comment (yongsokheng) [reply app/models/user_notification.rb]: tae trov brer chren kanleng, trov viet dor del dor del cheng doch ot sov laor
`scope :load_notifications, ->limit{limit(limit).preload notification: :user}`
viet cheng mech dae?

File: app/models/notification.rb (hunk 2 above)
Comment (NgengChhengkim) [reply app/models/notification.rb]: cheng co ban

File: app/models/user_notification.rb (hunk 5 above)
Comment (NgengChhengkim) [reply app/models/user_notification.rb]: hmmm. no cung mot loai. chinh la trang thai.

Comment (yongsokheng) [reply app/models/user_notification.rb]: ok. cheng kor ban

File: app/models/user_notification.rb (hunk 4 above)
Comment (NgengChhengkim) [reply app/models/user_notification.rb]: dak knong scope kert ta. sak merl merl

Comment (NgengChhengkim) [reply app/models/user_notification.rb]: limit, find_by ke ort sov oy dak knong scope doy sa te ort sov yok tov brer kert. limit ort doch knea.  find_by doch where te where ort limit ach hao brer pel na co ban

File: app/models/user_notification.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class UserNotification < ApplicationRecord
+  belongs_to :notification
+  belongs_to :user
+
+  scope :load_notifications, ->limit{limit(limit).preload notification: :user}
```
Comment (NgengChhengkim) [inline app/models/user_notification.rb]: `number{limit(number)`

**Pull request merged**